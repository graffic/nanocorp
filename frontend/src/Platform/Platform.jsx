import { Link } from 'react-router-dom'
import React from 'react'

const Platform = ({ platform }) => (
  <div>
    <code>{JSON.stringify(platform)}</code>
    <Link to='/'>Back</Link>
  </div>
)

export default Platform
