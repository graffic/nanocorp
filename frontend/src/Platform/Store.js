import { observable } from 'mobx'
import client from '../client'

class Store {
  @observable platform = { }
  @observable error = false
  @observable hasData = false

  get (campaignId, platformType) {
    this.hasData = false
    this.error = false

    return client.request(`
query Campaign($campaignId: Int!, $platformType: PlatformType!) {
  campaign(id: $campaignId) {
    platform(type: $platformType) {
      type
      status
      total_budget
      remaining_budget
      start_date
      end_date
      target_audiance {
        languages
        genders
        age_range
        locations
        interests
        KeyWords
      }
      creatives {
        headers
        description
        url
        image
      }
      insights{
        impressions
        clicks
        website_visits
        nanos_score
        cost_per_click
        click_through_rate
        advanced_kpi_1
        advanced_kpi_2
      }
    }
  }
}`, { campaignId, platformType })
      .then(({ campaign: { platform } }) => {
        this.platform = platform
        this.hasData = true
      })
      .catch(() => {
        this.error = true
      })
  }
}

export default Store
