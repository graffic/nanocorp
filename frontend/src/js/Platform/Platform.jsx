import React from 'react'

const Platform = ({ platform }) => (<p>{JSON.stringify(platform)}</p>)

export default Platform
