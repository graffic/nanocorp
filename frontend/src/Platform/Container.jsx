import React, { Component } from 'react'
import Store from './Store'
import Platform from './Platform'
import { observer } from 'mobx-react'

const store = new Store()

/**
 * Handles the retrieval of a platform when the component is mounted,
 * plus errors and loading screen.
 */
@observer
class Container extends Component {
  componentDidMount () {
    const { campaignId, platformType } = this.props.match.params
    store.get(parseInt(campaignId), platformType)
  }

  render () {
    if (store.error) return (<h1>Error Loading</h1>)
    if (!store.hasData) return (<h3>Loading...</h3>)
    return (<Platform platform={store.platform} />)
  }
}

export default Container
