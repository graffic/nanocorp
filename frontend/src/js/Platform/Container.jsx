import React from 'react'
import Platform from './Platform'
import { Link } from 'react-router-dom'

const Container = ({ match: { params } }) => <>
  <Platform platform={params} />
  <Link to='/'>back</Link>
</>

export default Container
