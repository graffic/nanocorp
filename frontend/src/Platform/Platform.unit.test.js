import Platform from './Platform'
import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'

const platform = {
  'status': 'Delivering',
  'total_budget': 40,
  'remaining_budget': 12,
  'start_date': 1530568800000,
  'end_date': 1532901600000,
  'target_audiance': {
    'languages': ['FR', 'EN', 'DE'],
    'genders': ['M', 'F'],
    'age_range': [20, 66],
    'locations': [
      'France',
      'Germany'
    ],
    'interests': [
      'Docker',
      'Kubernates'
    ]
  },
  'creatives': {
    'header': 'DevOps Made Easy, We Take care of the heavy lifting for you',
    'description': 'DOP SuperHero is where all DevOps is going to happen in the future, join the revolution today!',
    'url': 'https://example.io',
    'image': 'img1.jpg'
  },
  'insights': {
    'impressions': 4503,
    'clicks': 328,
    'nanos_score': 5.7,
    'cost_per_click': 0.88,
    'click_through_rate': 0.09,
    'advanced_kpi_1': 44.5,
    'advanced_kpi_2': 0.0023
  }
}

test('Renders', () => {
  const renderer = new ShallowRenderer()
  const result = renderer.render(<Platform platform={platform} />)
  expect(result).toMatchSnapshot()
})
