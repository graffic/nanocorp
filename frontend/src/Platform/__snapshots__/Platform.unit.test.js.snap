// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders 1`] = `
<div>
  <Link
    replace={false}
    to="/"
  >
    &lt; Back to campaign list
  </Link>
  <React.Fragment>
    <h1>
      Platform: 
    </h1>
    <ul>
      <li>
        Status: 
        Delivering
      </li>
      <li>
        Total budget: 
        40
      </li>
      <li>
        Remaining budget: 
        12
      </li>
      <li>
        Start: 
        Mon, 02 Jul 2018 22:00:00 GMT
      </li>
      <li>
        End: 
        Sun, 29 Jul 2018 22:00:00 GMT
      </li>
    </ul>
  </React.Fragment>
  <React.Fragment>
    <h2>
      Target Audience
    </h2>
    <ul>
      <li>
        Languages: 
        FR, EN, DE
      </li>
      <li>
        Genders: 
        M, F
      </li>
      <li>
        Age range: 
        20
         to 
        20
      </li>
      <li>
        Locations: 
        France, Germany
      </li>
      <li>
        Interests: 
        Docker, Kubernates
      </li>
      <li>
        Keywords: 
        --
      </li>
    </ul>
  </React.Fragment>
  <React.Fragment>
    <h2>
      Creatives
    </h2>
    <ul>
      <li>
        Headers: 
        --
      </li>
      <li>
        Description: 
        DOP SuperHero is where all DevOps is going to happen in the future, join the revolution today!
      </li>
      <li>
        <img
          src="undefinedimg1.jpg"
        />
      </li>
    </ul>
  </React.Fragment>
  <React.Fragment>
    <h2>
      Insights
    </h2>
    <ul>
      <li>
        Impressions: 
        4503
      </li>
      <li>
        Clicks: 
        328
      </li>
      <li>
        Website visits: 
      </li>
      <li>
        Nanos score: 
        5.7
      </li>
      <li>
        CPC: 
        0.88
      </li>
      <li>
        CTR: 
        0.09
      </li>
      <li>
        KPI 1: 
        44.5
      </li>
      <li>
        KPI 2: 
        0.0023
      </li>
    </ul>
  </React.Fragment>
</div>
`;
